USER_GUIDE.md

```diff
@@ -132,6 +132,54 @@ var response = lowLevelClient.Search<SearchResponse<Tweet>>("mytweetindex", Post
 }));
 ```
 
+## [OpenSearch.Net.Auth.AwsSigV4](src/OpenSearch.Net.Auth.AwsSigV4)
+
+An implementation of AWS SigV4 request signing for performing IAM authentication against the managed Amazon OpenSearch Service.
+It can be used with both the low-level OpenSearch.Net client as well as the higher-level OpenSearch.Client client.
+
+### Getting Started
+Include OpenSearch.Net.Auth.AwsSigV4 along with your preferred client in your .csproj file.
+```xml
+<Project>
+  ...
+  <ItemGroup>
+    <PackageReference Include="OpenSearch.Client" Version="1.0.0" />
+    <PackageReference Include="OpenSearch.Net.Auth.AwsSigV4" Version="1.0.0" />
+  </ItemGroup>
+</Project>
+```
+
+### Connecting
+The only wiring required is to use the `AwsSigV4HttpConnection` implementation of `IConnection`.
+
+**With OpenSearch.Client**
+```csharp
+var endpoint = new Uri("https://example-aaabbbcccddd111222333.us-east-1.es.amazonaws.com");
+var connection = new AwsSigV4HttpConnection();
+var config = new ConnectionSettings(endpoint, connection);
+var client = new OpenSearchClient(config);
+```
+
+**With OpenSearch.Net**
+```csharp
+var endpoint = new Uri("https://example-aaabbbcccddd111222333.us-east-1.es.amazonaws.com");
+var connection = new AwsSigV4HttpConnection();
+var config = new ConnectionConfiguration(endpoint, connection);
+var client = new OpenSearchLowLevelClient(config);
+```
+
+### Configuring Region & Credentials
+By default, `AwsSigV4HttpConnection` will use the same default logic as the [AWS SDK for .NET](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/creds-assign.html) to determine the credentials and region to use.
+However, you may explicitly specify one or both to override this logic, for example if you need to assume another IAM role first:
+```csharp
+var credentials = new AssumeRoleAWSCredentials(
+				FallbackCredentialsFactory.GetCredentials(),
+				"arn:aws:iam::123456789012:role/my-open-search-ingest-role",
+				"my-ingest-application");
+var region = RegionEndpoint.APSoutheast2;
+var connection = new AwsSigV4HttpConnection(credentials, region);
```

Comment:
For completeness add the lines that create a client.

Next, many users have an endpoint and a region, but set credentials via AWS_ env variables. Document how to pass a region without credentials here and check whether this works:


```
set AWS_ACCESS_KEY_ID=
set AWS_SECRET_ACCESS_KEY=
set AWS_SESSION_TOKEN=
```

```
connect = new ... (host, region)
```

Maybe useful as another example?

Reply:
I've added further documentation in the direction of your comment here